packages/mermaid/src/dagre-wrapper/blockArrowHelper.js

```diff
@@ -0,0 +1,218 @@
+const expandAndDeduplicateDirections = (directions) => {
+  const uniqueDirections = new Set();
+
+  for (const direction of directions) {
+    switch (direction) {
+      case 'x':
+        uniqueDirections.add('right');
+        uniqueDirections.add('left');
+        break;
+      case 'y':
+        uniqueDirections.add('up');
+        uniqueDirections.add('down');
+        break;
+      default:
+        uniqueDirections.add(direction);
+        break;
+    }
+  }
+
+  return uniqueDirections;
+};
+export const getArrowPoints = (directions, bbox, node) => {
+  const ud = expandAndDeduplicateDirections(directions);
+
+  // console.log('block_arrow abc123', node.id, node.directions, ud);
+
+  const f = 2;
+  const h = bbox.height + 2 * node.padding;
+  const m = h / f;
+  const w = bbox.width + 2 * m + node.padding;
+  const p = node.padding / 2;
```

Comment:
nit: Can you use the rename symbol function to use proper names here? Would help with readability and maintenance.